### PR TITLE
Adding class for gas equations

### PR DESCRIPTION
**Pull request by seanbaccas**

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -12,6 +12,11 @@ target_link_libraries(Utils Logging)
 
 add_library(Parameters Parameters.cpp Parameters.h)
 
+add_library(Gas Gas.cpp Gas.h)
+target_include_directories(Gas PUBLIC "${PROJECT_BINARY_DIR}") # we need this for Config.h
+target_link_libraries(Gas Version)
```
Comment (mladenivkovic) [inline src/CMakeLists.txt:17]: why do you need Version here?

File: src/Constants.h [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+#define GAMMA (5. / 3.)
+
+
+
+/* JUST PUTTING THESE IN SO IT COMPILES */
+#ifdef USE_AS_RIEMANN_SOLVER
+/* set the "small" values to actually zero, so that only correct vacuum sates
+ * are recognized as such */
+#define SMALLRHO 0.
```
Comment (mladenivkovic) [inline src/Constants.h:11]: maybe we should be using `constexpr` instead of macro definitions....

File: src/Constants.h [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+#define SMALLU 0.
+#define SMALLP 0.
+#else
+/* cheat for stability in Godunov type finite volume schemes*/
+#define SMALLRHO 1e-6
+#define SMALLU 1e-6
+#define SMALLP 1e-6
+#endif
+
+static const float GM1 = GAMMA - 1.;
```
Comment (mladenivkovic) [inline src/Constants.h:21]: same here: maybe should be constexpr

File: src/Gas.cpp [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+void      IdealGas::PrimitiveState::setRho(const Precision val){
+  rho = val;
+}
+
+Precision IdealGas::PrimitiveState::getRho() const{
+  return rho;
+}
+
+void      IdealGas::PrimitiveState::setU(int index, const Precision val){
+  // bounds checking? why not.... if we have time
```
Comment (mladenivkovic) [inline src/Gas.cpp:38]: what bounds are you checking here?

**Review by mladenivkovic — COMMENTED** [review]
Can you run the formatting script on this too?

File: src/Gas.cpp (hunk 4 above)
Comment (seanbaccas) [reply src/Gas.cpp:38]: was checking that we didn't pass invalid index. should be fine to remove

**Comment by mladenivkovic** [comment]
Looks good!

A few minor nitpicks on my side:

- we should probably change the velocity `u` to `v`. Don't remember why I decided to stick with that, but it's probably not a great idea. `u` is usually used for the specific internal energy density.
- I'm not a big fan of the naming `Precision`. I'd prefer it to be something that indicates that this is a floating point number.
- I think we should probably move the `primitiveState` and the `ConservedState` classes into their separate files. Do clean C++, not however I was abusing C in the original project.

But all of these can wait for some other MR. I'd prioritize getting this thing to run in the first place. Unless you disagree, I'll put this into the ToDo-list. I.e. open issues for each of these.

So merging in for now.

**Review by mladenivkovic — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**